// zahir-api/lib/Muslim.js
const fetch = require('node-fetch')
const asmaul = 'https://raw.githubusercontent.com/Zhirrr/islamic-rest-api-indonesian/main/data/dataAsmaulHusna.json'
const tahlilan = 'https://raw.githubusercontent.com/Zhirrr/islamic-rest-api-indonesian/main/data/dataTahlil.json'
const kursi = 'https://raw.githubusercontent.com/Zhirrr/islamic-rest-api-indonesian/main/data/dataAyatKursi.json'
const wiridd = 'https://raw.githubusercontent.com/Zhirrr/islamic-rest-api-indonesian/main/data/dataWirid.json'

module.exports.AsmaulHusna = function AsmaulHusna() {
	return new Promise((resolve, reject) => {
		fetch(asmaul)
		.then(res => res.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Tahlil = function Tahlil() {
	return new Promise((resolve, reject) => {
		fetch(tahlilan)
		.then(res => res.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.AyatKursi = function AyatKursi() {
	return new Promise((resolve, reject) => {
		fetch(kursi)
		.then(res => res.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Wirid = function Wirid() {
	return new Promise((resolve, reject) => {
		fetch(wiridd)
		.then(res => res.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.JadwalShalat = function JadwalShalat(kota) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://raw.githubusercontent.com/Zhirrr/Zhirrr-Database/main/adzan/${kota}/2021/03.json`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.KisahNabi = function KisahNabi(nabi) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://raw.githubusercontent.com/Zhirrr/My-SQL-Results/main/kisahnabi/${nabi}.json`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.DoaHarian = function DoaHarian() {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://raw.githubusercontent.com/Zhirrr/My-SQL-Results/main/data/dataDoaHarian.json`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.BacaanShalat = function BacaanShalat() {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://raw.githubusercontent.com/Zhirrr/My-SQL-Results/main/data/dataBacaanShalat.json`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Quran = function Quran(surah, ayat) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://alquran-apiii.vercel.app/surah/${surah}/${ayat}`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Hadits = function Quran(kitab, nomor) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://hadits-api-zhirrr.vercel.app/books/${kitab}/${nomor}`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}
