// zahir-api/lib/TextMaker.js
const fetch = require('node-fetch')

module.exports.Glitch = function Glitch(text, text2) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://vip-api-zahirr.herokuapp.com/api/textmaker?text=${text}&text2=${text2}&theme=glitch&apikey=zahirgans`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Google = function Google(text, text2, text3) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://vip-api-zahirr.herokuapp.com/api/textmaker?text=${text}&text2=${text2}&text3=${text3}&theme=google-suggestion&apikey=zahirgans`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Neon = function Neon(text) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://vip-api-zahirr.herokuapp.com/api/textmaker/metallic?text=${text}&theme=neon&apikey=zahirgans`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Glow = function Glow(text) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://vip-api-zahirr.herokuapp.com/api/textmaker/metallic?text=${text}&theme=glow&apikey=zahirgans`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Summer = function Summer(text) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://vip-api-zahirr.herokuapp.com/api/textmaker/alam?text=${text}&theme=summer&apikey=zahirgans`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.Flower = function Flower(text) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://vip-api-zahirr.herokuapp.com/api/textmaker/alam?text=${text}&theme=flower&apikey=zahirgans`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.CoffeeCup = function CoffeCup(text) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://vip-api-zahirr.herokuapp.com/api/textmaker/senja?text=${text}&theme=coffee-cup&apikey=zahirgans`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}

module.exports.CoffeeCup2 = function CoffeCup2(text) {
	return new Promise((resolve, reject) => {
		fetch(encodeURI(`https://vip-api-zahirr.herokuapp.com/api/textmaker/senja?text=${text}&theme=coffee-cup2&apikey=zahirgans`))
        .then(response => response.json())
		.then(resolve)
		.catch(reject)
	})
}
